Bugzilla/API/V1/Github.pm

```diff
@@ -173,4 +340,49 @@ sub verify_signature {
   return secure_compare($expected_signature, $received_signature) ? 1 : 0;
 }
 
+sub _set_status_flag {
+  my ($self, $bug, $ref) = @_;
+
+  use Bugzilla::Logging;
+  DEBUG("ref: $ref");
+
+  my ($branch) = $ref =~ /refs\/heads\/(.*)$/;
+  return if !$branch;
+
+  DEBUG("branch: $branch");
+
+  my $version;
+  if ($branch eq 'main' || $branch eq 'master') {
+    my $versions  = fetch_product_versions('firefox');
+    return if (!%$versions || !exists $versions->{FIREFOX_NIGHTLY});
+    ($version) = split /[.]/, $versions->{FIREFOX_NIGHTLY};
+  }
+  else {
+    ($version) = $branch =~ /^releases_v(\d+)$/;
+  }
+  return if !$version;
```

Comment:
Can you add more comments describing this?

Reply:
Sure